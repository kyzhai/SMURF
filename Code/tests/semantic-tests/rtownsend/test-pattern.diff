3c3
< 		i [a, [2], [3]] :: Unknown = 5
---
> 		f (a : []) (b : c) (d : e : f : [] : []) (3 : 2 : _ : 4 : _) :: [Int] -> [Int] -> [a] -> [Int] -> Int = 5
4a5,10
> 		ID: a  :: Int
> 		ID: b  :: Int
> 		ID: c  :: [Int]
> 		ID: d  :: a
> 		ID: e  :: a
> 		ID: f  :: a
8c14
< 		i [[1, 2], c] :: Unknown = 4
---
> 		g (true : false : []) (true : false : [true, false]) ([3, 4, 5] : []) :: [Bool] -> [Bool] -> [[Int]] -> Int = 1
13c19
< 		i [a, b, c] :: Unknown = 3
---
> 		h ([3, 4, 5] : [[1]]) :: [[Int]] -> Int = 2
18c24
< 		h [3, 4, 5] : [[1]] :: Unknown = 2
---
> 		i ([a, b, c]) :: [[Int]] -> Int = 3
19a26,28
> 		ID: a  :: [Int]
> 		ID: b  :: [Int]
> 		ID: c  :: [Int]
23c32
< 		g true : false : [] true : false : [true, false] [3, 4, 5] : [] :: Unknown = 1
---
> 		i ([[1, 2], c]) :: [[Int]] -> Int = 4
24a34
> 		ID: c  :: [Int]
28c38
< 		f a : [] b : c d : e : f : [] : [] 3 : 2 : _ : 4 : _ :: Unknown = 5
---
> 		i ([a, [2], [3]]) :: [[Int]] -> Int = 5
29a40
> 		ID: a  :: [Int]
34,39d44
< 	ID: main  :: Unknown
< 	ID: i [a, [2], [3]] :: [[Int]] -> Int
< 	ID: i [[1, 2], c] :: [[Int]] -> Int
< 	ID: i [a, b, c] :: [[Int]] -> Int
< 	ID: h [3, 4, 5] : [[1]] :: [[Int]] -> Int
< 	ID: g true : false : [] true : false : [true, false] [3, 4, 5] : [] :: [Bool] -> [Bool] -> [[Int]] -> Int
40a46,51
> 	ID: g true : false : [] true : false : [true, false] [3, 4, 5] : [] :: [Bool] -> [Bool] -> [[Int]] -> Int
> 	ID: h [3, 4, 5] : [[1]] :: [[Int]] -> Int
> 	ID: i [a, b, c] :: [[Int]] -> Int
> 	ID: i [[1, 2], c] :: [[Int]] -> Int
> 	ID: i [a, [2], [3]] :: [[Int]] -> Int
> 	ID: main  :: Unknown
42a54,55
> 	ID: head h : t :: [a] -> a
> 	ID: tail h : t :: [a] -> [a]
