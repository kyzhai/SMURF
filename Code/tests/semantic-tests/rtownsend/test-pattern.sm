/* Testing cons pattern matching */
f :: [Int] -> [Int] -> [a] -> [Int] -> Int
f a:[] b:c d:e:f:[]:[] 3:2:_:4:_ = 5

g :: [Bool] -> [Bool] -> [[Int]] -> Int

g (True:((False)):[]) True:False:[True, False] [3,4,5]:[] = 1

h :: [[Int]] -> Int
h [3,4,5]:[[1]] = 2

i :: [[Int]] -> Int
i [a,b,c] = 3
i [[1,2],c] = 4
i [a, [2], [3]] = 5

main = []
