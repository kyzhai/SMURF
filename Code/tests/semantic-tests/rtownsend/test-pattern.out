Program: 
	SFuncdec: 
		i [a, [2], [3]] :: Unknown = 5
	New Scope: 
		
	
	SFuncdec: 
		i [[1, 2], c] :: Unknown = 4
	New Scope: 
		
	
	SFuncdec: 
		i [a, b, c] :: Unknown = 3
	New Scope: 
		
	
	SFuncdec: 
		h [3, 4, 5] : [[1]] :: Unknown = 2
	New Scope: 
		
	
	SFuncdec: 
		g true : false : [] true : false : [true, false] [3, 4, 5] : [] :: Unknown = 1
	New Scope: 
		
	
	SFuncdec: 
		f a : [] b : c d : e : f : [] : [] 3 : 2 : _ : 4 : _ :: Unknown = 5
	New Scope: 
		
	
	SMain: []
Global Scope: 
	ID: main  :: Unknown
	ID: i [a, [2], [3]] :: [[Int]] -> Int
	ID: i [[1, 2], c] :: [[Int]] -> Int
	ID: i [a, b, c] :: [[Int]] -> Int
	ID: h [3, 4, 5] : [[1]] :: [[Int]] -> Int
	ID: g true : false : [] true : false : [true, false] [3, 4, 5] : [] :: [Bool] -> [Bool] -> [[Int]] -> Int
	ID: f a : [] b : c d : e : f : [] : [] 3 : 2 : _ : 4 : _ :: [Int] -> [Int] -> [a] -> [Int] -> Int
	ID: print x :: a -> a
	ID: random  :: Int

